app/models/host.rb

```diff
@@ -90,7 +90,7 @@ class Host < ApplicationRecord
 
   scope :with_failed_rules_count, lambda {
     sq = Host.left_outer_joins(test_result_profiles: :rule_results)
-             .where(rule_results: { result: RuleResult::FAILED })
+             .where("rule_results.result IN (?) OR rule_results.id IS NULL", RuleResult::FAILED)
```

Comment:
You can do this in SQL :pray: look at how `or` works

Reply:
Right, I considered that but it was pretty long but I can switch that 👍

Reply:
I think you don't need more in the OR than `Host.where(rule_results: { id: nil })`

Reply:
I'll try that. I was going to do something like this:
```
join_trp_rr = Host.left_outer_joins(test_result_profiles: :rule_results)
    sq = join_trp_rr.where(rule_results: { result: RuleResult::FAILED })
                    .or(join_trp_rr.where(rule_results: { id: nil }))
                    .select(arel_table[:id].as('id'), RuleResult.arel_table[:result].count.as('rules_failed'))
                    .group('hosts.id')
```

Reply:
`Host.where(rule_results: { id: nil })` worked so I'll use that 👍

Reply:
You don't need to overcomplicate the query inside the `or`, it is not a "real" query, just something that can be merged with the main one.